### PR TITLE
update ace to v1.2.9; fix ts build;

### PR DESCRIPTION
**Pull request by filipsch**
- update `aceTag` to `v1.2.8`
- updated `build/update-ts.js` to follow new structure of `DefinitivelyTyped` repo.

### CODE REVIEW

**Comment by thlorenz** [comment]
Thanks @filipsch.
Could I get some more @collaborators to review this? On quick skim LGTM.

File: ext/searchbox.js [hunk 1]
```diff
@@ -184,19 +200,27 @@ var SearchBox = function(editor, range, showReplaceForm) {
 (function() {
     this.setEditor = function(editor) {
         editor.searchBox = this;
-        editor.container.appendChild(this.element);
+        editor.renderer.scroller.appendChild(this.element);
         this.editor = editor;
     };
+    
+    this.setSession = function(e) {
+        debugger
```
Comment (gokmen) [inline ext/searchbox.js]: that seems weird, it's introduced with https://github.com/ajaxorg/ace/pull/3329/files#diff-2b1fc8367171d51dd3148748a50accb2R88 on Ace repo 29 days ago. maybe we should let them know first fix there and merge after the change.

Comment (gokmen) [inline ext/searchbox.js]: created a PR for that https://github.com/ajaxorg/ace/pull/3358 fyi.

Comment (thlorenz) [inline ext/searchbox.js]: OK let's wait for that to go through and then run the upgrade again.
Don't want to pull anything in that introduces an obvious bug from the upstream code base.

Comment (gokmen) [inline ext/searchbox.js]: it's merged on upstream, if @filipsch can update the code we are good to go I believe 👍

Comment (filipsch) [inline ext/searchbox.js]: I guess I'll have to wait for ajaxorg to make a new release of ace, right?

Comment (thlorenz) [inline ext/searchbox.js]: Yeah better since we don't want to depend on an unreleased version.

**Comment by pankitgami** [comment]
When will you merge this request? As `ace` is on 1.2.8 and this package is on 1.2.6. And there are lots of changes between both the release in terms of worker, theme, mode etc.

**Comment by filipsch** [comment]
@gokmen noticed that the people from ace merged in a bug (a `debugger` call) and this bug is in 1.2.8 of ace. A fix of this has been merged, but we're still waiting for the people of ace to bump their version, so we can start brace from a stable version that doesn't have the bug.

**Comment by pankitgami** [comment]
bug_brace
This version of the brace has a bug in php-worker.

`editor.js:3609 Could not load worker TypeError: Cannot read property 'split' of undefined`

**Comment by shepmaster** [comment]
Is there a link to an issue requesting a new ace release that someone could provide?

FWIW, I'd be OK with using Ace 1.2.8 even *with* the debugger statement, especially considering there are bugs that impact my downstream users fixed in 1.2.8.

To my knowledge, the `debugger` statement will only have an effect when the user has the browser inspector open, and in that case you presumably understand what you are doing and how to continue past a rogue `debugger`.

From another POV, if brace upgrades to 1.2.8, then more people can experience any issue that this causes and then check in with upstream to get it fixed / released.

**Comment by shepmaster** [comment]
[Ace 1.2.9 has been released](https://github.com/ajaxorg/ace/releases/tag/v1.2.9)! Now with one less debugger statement 😺

**Comment by filipsch** [comment]
@shepmaster I'll update the PR then to depend on v1.2.9, and see if the builds still work.

**Comment by filipsch** [comment]
Done, but my new commit and push to the `pr` branch of `datacamp/brace` fork is not coming through for some reason.

**Comment by shepmaster** [comment]
Huh; I see the same behavior. *Maybe* it's just some GitHub slowness?

**Comment by filipsch** [comment]
@shepmaster Okay cool; we're ready then. @gokmen can this be merged in now, and can a new release be made at the same time? Thanks!

**Comment by gokmen** [comment]
it's good to go for me 👍, on you @thlorenz

**Comment by filipsch** [comment]
@thlorenz ping!

**Comment by thlorenz** [comment]
Sorry, just seeing this now. LGTM

@filipsch made you collaborator.

Please merge to master following [this guide](https://gist.github.com/thlorenz/f7df0daf13c6f6c1fa84).
I will then version and publish to npm.

Thanks.

Obviously @gokmen if you see this first and want to merge, fine by me.

Just ping me when done and I'll publish a new minor to npm.

**Comment by filipsch** [comment]
@thlorenz done!

**Comment by thlorenz** [comment]
@filipsch getting test errors (did you run `npm test` and open devtools?).
I don't see any error annotations in the editors, seems like something not working with the `worker` scripts.

I have 10/10 tests failing.

If we cannot solve this today we need to revert the commits from master.
I thought we had checked the tests before merging.

**Comment by thlorenz** [comment]
OK, confirmed that before these commits everything works fine.

```
> git reset --hard a417c2e
HEAD is now at a417c2e Fix link to DefinitelyTyped definition in README

> npm ts
```

**Comment by filipsch** [comment]
@thlorenz will look into it before EOD.

**Comment by thlorenz** [comment]
OK I force pushed to master, reverting the commits (hoping no one else but the two of us updated from master at this point).
The errors are a regression we cannot have, one main point of brace is that the worker scripts are inlined and annotations just work out of the box.

Sorry to do this, but I feel this is better to have reverted these and now we can calmly figure out why things broke.

**Comment by filipsch** [comment]
@thlorenz completely understand; I should have run the test scripts before; my bad.

**Comment by thlorenz** [comment]
No problem (we caught it in time). Probably best to open another PR with the changes as I cannot reopen this one.

**Comment by shepmaster** [comment]
> I force pushed to master, reverting the commits

For the future, you can just revert the merge. GitHub should even have a button:

image

**Comment by thlorenz** [comment]
@shepmaster I know you can do that, but that adds extra commits.
The fact that this was up for only 30 mins made me think no one would get hurt if I just force push ;)

**Comment by LeviRosol** [comment]
Is this still going to happen? I was looking forward to the update.

**Comment by filipsch** [comment]
@LeviRosol it's still on my todo list, but I'm pretty swamped at my daytime job currently. When doing this PR, I didn't dive into the `brace` codebase itself, and basically just updated the build scripts. Therefore, I guess it will take a pretty long time for me to figure out where and why the tests now fail. I have no idea how long this is going to take.

**Comment by LeviRosol** [comment]
no worries. I assumed as much. If I can find some time I'll try to lend a hand.

**Pull request merged**